### PR TITLE
Ssh rsa deprecation docs adjustments

### PR DESCRIPTION
**Pull request by anurgatin**
We have announced `SSH-RSA` cipher deprecation. This PR suppose to adjust public docs we provide about setting ssh key for ado git experience according to incomming changes

see: https://devblogs.microsoft.com/devops/ssh-rsa-deprecation/

### CODE REVIEW

**Comment by bjanousek** [comment]
I'll Czech this tomorrow.

Bohdan

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Review by bjanousek — APPROVED** [review]
The changes looks good now.

**Comment by bjanousek** [comment]
#sign-off

**Comment by bjanousek** [comment]
@Jak-MS - can you merge this PR. It's ready to go

**Comment by bjanousek** [comment]
#sign-off

File: docs/repos/git/use-ssh-keys-to-authenticate.md [hunk 1]
```diff
@@ -268,48 +268,98 @@ Associate the public key generated in the previous step with your user ID.
...
+   ```
+
+   Modify your SSH config to downgrade your security settings for Azure DevOps by adding the following to your `~/.ssh/config` (`%UserProfile%\.ssh\config` on Windows) file:
+
+   ```
+   Host ssh.dev.azure.com vs-ssh.visualstudio.com
+     MACs +hmac-sha2-512,+hmac-sha2-256
+   ```
+
+4. **No matching key exchange method**
```
Comment (Court72) [inline docs/repos/git/use-ssh-keys-to-authenticate.md:348]:
```suggestion
- **No matching key exchange method**
```

File: docs/repos/git/use-ssh-keys-to-authenticate.md [hunk 2]
```diff
@@ -268,48 +268,98 @@ Associate the public key generated in the previous step with your user ID.
...
+
+   ```
+   Host ssh.dev.azure.com vs-ssh.visualstudio.com
+     HostkeyAlgorithms +ssh-rsa
+   ```
+
+  > [!IMPORTANT]
+  > OpenSSH deprecated the `ssh-rsa` public key signature algorithm in [version 8.2](https://www.openssh.com/txt/release-8.2) and disabled it by default in [version 8.8](https://www.openssh.com/txt/release-8.8).
+
+3. **No matching MAC**
```
Comment (Court72) [inline docs/repos/git/use-ssh-keys-to-authenticate.md:335]:
```suggestion
- **No matching MAC**
```

File: docs/repos/git/use-ssh-keys-to-authenticate.md [hunk 3]
```diff
@@ -268,48 +268,98 @@ Associate the public key generated in the previous step with your user ID.
...
-Modify your SSH config to downgrade your security settings for Azure DevOps by adding the following to your `~/.ssh/config` (`%UserProfile%\.ssh\config` on Windows) file:
+   ```output
+   You’re using ssh-rsa that is unsupported. Please use rsa-sha2-256 or rsa-sha2-512 instead. For more details see https://devblogs.microsoft.com/devops/ssh-rsa-deprecation.
+   ```
 
-```
-Host ssh.dev.azure.com vs-ssh.visualstudio.com
-  KexAlgorithms +diffie-hellman-group-exchange-sha256,+diffie-hellman-group14-sha1,+diffie-hellman-group1-sha1
-```
+   You may have previously modified your SSH config to downgrade your security settings for Azure DevOps by adding the following to your `~/.ssh/config` (`%UserProfile%\.ssh\config` on Windows) file:
```
Comment (Court72) [inline docs/repos/git/use-ssh-keys-to-authenticate.md:306]:
```suggestion
   You may have previously modified your SSH config to downgrade your security settings for Azure DevOps by adding the following to your `~/.ssh/config` (`%UserProfile%\.ssh\config` on Windows) file:
```

File: docs/repos/git/use-ssh-keys-to-authenticate.md [hunk 4]
```diff
@@ -268,48 +268,98 @@ Associate the public key generated in the previous step with your user ID.
...
-```
-Host ssh.dev.azure.com vs-ssh.visualstudio.com
-  MACs +hmac-sha2-512,+hmac-sha2-256
-```
+**A:** There are multiple different problems that you may experience:
 
-```Output
-Unable to negotiate with <IP> 22: no matching key exchange method found. Their offer: diffie-hellman-group1-sha1,diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha256
-```
+1. **Use of unsupported SSH-RSA**
```
Comment (Court72) [inline docs/repos/git/use-ssh-keys-to-authenticate.md:300]:
```suggestion
- **Use of unsupported SSH-RSA**
```

**Comment by bjanousek** [comment]
Hello Courtney,

as per the message in Teams to your MS account:

When merging the changes for this PR there are issue with some texts that does not render correctly. When I've checked the preview in "main" branch I see the "important" texts are not rendered OK after you've changed the numbers to bullet points. Can you fix that?



Specifically these are the two

[cid:a5d0654e-aa33-4c73-9a2b-731d1a619ce1]

Thanks,

Bohdan Janousek
Senior Program Manager
Azure DevOps
***@***.***

[Microsoft Logo]
________________________________

**Pull request merged**